src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -559,12 +561,26 @@ def reset_service_principal_credential(name, secret=None, years=1):
 
     client.applications.patch(app.object_id, app_create_param)
 
-    _build_output_content(name, secret, client.config.tenant_id)
-
-def _build_output_content(sp_name, secret, tenant):
-    logger.warning("Service principal has been configured.")
-    logger.warning("  id(client_id):           " + sp_name)
-    logger.warning("  password(client_secret): " + secret)
+    _build_output_content(name, app.app_id, secret, client.config.tenant_id)
+
+def _build_output_content(sp_name, app_id, secret, tenant):
+    #this is a fairly special convinience command that the result is not associated
+    #with one api response, so it controls its own output format for easy scripting
+    #and also ensures warning (to stderr) goes after the stdout.
+    from azure.cli.core.application import APPLICATION
+
+    if APPLICATION.configuration.use_json_formatter():
+        result = {
+            'client_id': app_id,
+            'client_secret': secret,
+            'sp_name': sp_name,
+        }
+        print(json.dumps(result, indent=2))
```

Comment:
Since main takes a stream parameter (which is supposed to represent where commands output their information), you cannot blindly use print since that wouldn't honor that setting..